### PR TITLE
[#1603] Redirect when trying to add GBM channel

### LINKED ISSUE #1603: Redirect when trying to add GBM Channel
### Current Behaviour

When clicking to add a GBM channel, I get a redirect to my last conversation

### Expected Behaviour

Either dialog to connect the channel, or a message showing the channel is not there

gbm-bug

### PR DESCRIPTION
**Pull request by AudreyKj**
closes #1603

- GBM modal box linked to the Google source from the docs

Screenshot 2021-04-22 at 13 18 00

### CODE REVIEW

File: frontend/ui/src/pages/Channels/MainPage/index.tsx [hunk 1]
```diff
@@ -113,7 +114,7 @@ const SourcesInfo: SourceInfo[] = [
     image: <GoogleAvatarIcon />,
     newChannelRoute: '/',
     channelsListRoute: '/',
-    configKey: 'sources-google',
+    configKey: '',
```
Comment (chrismatix) [inline frontend/ui/src/pages/Channels/MainPage/index.tsx:117]: Maybe I'm not getting it, but how can this field be empty? 🤔

Comment (AudreyKj) [reply frontend/ui/src/pages/Channels/MainPage/index.tsx:117]: yes I know it's counter-intuitive, but I had to remove Google's configKey in order to have the modal pop-up. 

the backend returns `sources-google: {enabled: true}` but we are not yet able to connect Google via the UI, so we want to show the modal box describing the necessary configuration to the user

Comment (AitorAlgorta) [reply frontend/ui/src/pages/Channels/MainPage/index.tsx:117]: then you should add the `configKey` and the backend shoud return `enabled: false`. @AudreyKj  @chrismatix  @lucapette

Comment (lucapette) [reply frontend/ui/src/pages/Channels/MainPage/index.tsx:117]: yeah @AudreyKj I'm afraid I misunderstod the problem here. We have an issue to fix this behaviour. But it won't work like @AitorAlgorta is mentioning. `enabled: false` will be returned when google is correctly configured _but_ for some reason is not running at the moment.

As of last week, the `/client.config` endpoint has a different contract. We return in the response only components that are present in the airy instance (which means they've been configured already). So the right condition for the flow of the channels new accounts pages is `source component is not null and enabled` to proceed with it.


Having said that: I would merge this anyway. And fix this back in #1599 (which I think is already being worked on by @paulodiniz)

File: frontend/ui/src/pages/Channels/MainPage/index.tsx [hunk 2]
```diff
@@ -160,7 +162,7 @@ const MainPage = (props: MainPageProps & RouteComponentProps) => {
               sourceInfo={infoItem}
               displayButton={!channelsBySource(infoItem.type).length}
               addChannelAction={() => {
-                if (config.components[infoItem.configKey] && config.components[infoItem.configKey].enabled) {
+                if (config.components[infoItem.configKey] && !config.components[infoItem.configKey].enabled) {
                   props.history.push(infoItem.newChannelRoute);
```
Comment (AudreyKj) [inline frontend/ui/src/pages/Channels/MainPage/index.tsx:166]: updated the check to reflect the endpoint change

**Review by AitorAlgorta — APPROVED** [review]
🚀

**Pull request merged**